### PR TITLE
MalformedRoundaboutCheck Use ComplexRoundabout

### PR DESCRIPTION
**Pull request by Bentleysb**
### Description:

With the addition of the CompleRoundabout class to Atlas, much of the logic in MalformedRoundaboutCheck is redundant. This removes the redundant logic and replaces it with calls to ComplexRoundabout.

### Potential Impact:
There should not be any downstream impact. 

### Unit Test Approach:

Added one unit test for the left driving countries configurable.

### Test Results:

Tested the check on 5 countries. The flags were the same before and after the changes.

### CODE REVIEW

**Review by nachtm — APPROVED** [review]
🎊

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java [hunk 1]
```diff
@@ -194,64 +86,38 @@ public boolean validCheckForObject(final AtlasObject object)
...
-        }
-
-        // Flag if any of the edges are not car navigable or master Edges, or the route does not
-        // form a closed loop.
-        if (roundaboutEdgeSet.stream()
-                .anyMatch(roundaboutEdge -> !HighwayTag.isCarNavigableHighway(roundaboutEdge)
-                        || !roundaboutEdge.isMasterEdge())
-                || !roundaboutEdges.start().inEdges().contains(roundaboutEdges.end()))
+        // Create a ComplexRoundabout based on object
+        final ComplexRoundabout complexRoundabout = new ComplexRoundabout((Edge) object,
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java:91]: :100

**Review by danielduhh — APPROVED** [review]
Love net the loss in code. Thanks @Bentleysb

File: src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java (hunk 1 above)
Comment (danielduhh) [reply src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java:91]: 💯

**Pull request closed without merging**